compiler-core/src/javascript/expression.rs

```diff
@@ -1357,12 +1357,16 @@ pub fn string(value: &str) -> Document<'_> {
 
 pub fn array<'a, Elements: IntoIterator<Item = Output<'a>>>(elements: Elements) -> Output<'a> {
     let elements = Itertools::intersperse(elements.into_iter(), Ok(break_(",", ", ")))
-        .collect::<Result<Vec<_>, _>>()?
-        .to_doc();
+        .collect::<Result<Vec<_>, _>>()?;
+    let (initial_break, trailing_comma) = if elements.is_empty() {
+        (nil(), nil())
+    } else {
+        (break_("", ""), break_(",", ""))
+    };
     Ok(docvec![
         "[",
-        docvec![break_("", ""), elements].nest(INDENT),
-        break_(",", ""),
+        docvec![initial_break, elements].nest(INDENT),
+        trailing_comma,
         "]"
```

Comment:
I think an easier way to go about this would be something like this:
```rs
if elements.is_empty() {
  Ok("[]".to_doc())
} else {
  Ok(docvec![
    "[",
    docvec![break_("", ""), elements].nest(INDENT),
    break_(",", ""),
    "]",
  ])
}
```
So it's easier to see that the list is never split when there's no elements

Reply:
I like this idea here, but I would like to move away from this for the formatter so that you can have comments in an empty list

Reply:
Good idea! It has been done.